src/test/java/com/google/api/codegen/util/NameTest.java

```diff
@@ -139,4 +139,22 @@ public void ambiguousAcronym() {
   public void ambiguousRepeatedAcronym() {
     System.out.println(Name.upperCamel("APIDogAPIAMName").toLowerUnderscore());
   }
+
+  @Test
+  public void upperCamelUpperAcronymsSeparate() {
+    Name name = Name.upperCamelKeepUpperAcronyms("IAM", "HTTP", "XML", "Dog");
+    Truth.assertThat(name.toLowerUnderscore()).isEqualTo("iam_http_xml_dog");
+    Truth.assertThat(name.toUpperUnderscore()).isEqualTo("IAM_HTTP_XML_DOG");
+    Truth.assertThat(name.toLowerCamel()).isEqualTo("iamHTTPXMLDog");
```

Comment:
Is this definitely what we want when calling toLowerCamel on a string that starts with an acronym?